### PR TITLE
[Doc] Add a cluster xgboost example for vm stack

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
This is adapted from the same example of the k8s stack.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:12]: `AWS cloud` -> `AWS`.

We should mention that similar steps can be used to deploy on GCP or Azure.
(Later, we can add an actual cluster config for GCP.)

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:15]: There was a PR that updated the XGBoost references. https://github.com/ray-project/ray/pull/27687/files
It's now
```
To learn more about using Ray's XGBoostTrainer, check out {ref}`the XGBoostTrainer documentation <train-gbdt-guide>`.
```

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:50]: nit
`the aws` -> `your AWS`

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:3]: Note the re-branding from this PR:
https://github.com/ray-project/ray/pull/27687/files
There might be a couple more spots that need updates.

**Comment by DmitriGekhtman** [comment]
Did you try running the workload?

**Comment by jjyao** [comment]
> Did you try running the workload?

Yes, I ran and it worked. In the future, we should look into how to automate it and test it as part of CI or release test.

**Review by cadedaniel — APPROVED** [review]
Nice work, thanks for doing this @jjyao. LGTM from a structuring/docs perspective

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:3]: nit: title should be
`# Ray AIR XGBoostTrainer on VMs`

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml:1]: nit:
`XGBoost-Ray workload` -> `Ray AIR XGBoostTrainer benchmark`

We should probably modify the k8s example yaml too.

**Review by DmitriGekhtman — APPROVED** [review]
Left a couple more naming nits, but otherwise lgtm!

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml
Comment (jjyao) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml:1]: I'm fine with that. I did this because it's the structured used by k8s stack. cc @DmitriGekhtman

Comment (jjyao) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml:1]: Actually it might be better to put all configs in `configs` folder similar to all code is in `doc_code` folder. This way, later we can test all configs in this folder through CI.

Comment (cadedaniel) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/configs/xgboost-benchmark.yaml:1]: True. I'm fine deferring it to later.

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/xgboost-benchmark.yaml
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/xgboost-benchmark.yaml:1]: thanks!

**Comment by DmitriGekhtman** [comment]
Should be good to merge once doc lint passes. I think we need ray-docs team approval as well.

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/ml-example.md:85]: don't you need to run ray job submit here?

**Review by richardliaw — APPROVED** [review]

**Comment by jjyao** [comment]
Failed tests are unrelated.

**Pull request merged**
